File: app/src/main/java/com/alexpaxom/workrhythm/ui/main/FillTaskDataFormActivity.java
package com.alexpaxom.workrhythm.ui.main;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.SeekBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.alexpaxom.workrhythm.R;
import com.alexpaxom.workrhythm.TimeFormatter;
import com.alexpaxom.workrhythm.model.Task;
import com.alexpaxom.workrhythm.viewmodel.TaskStorageViewModel;

import androidx.core.util.Consumer;

import java.util.GregorianCalendar;

public class FillTaskDataFormActivity extends AppCompatActivity {
    private TaskStorageViewModel viewModel;
    private EditText title;
    private EditText descr;
    private TextView priority;
    private Intent intentData;
    private Task fillingTask;
    private ImageButton deleteTaskButton;
    private SeekBar seekBarPriority;

    private EditText formatElapsedTime;
    private Button increaseElapsedTimeButton;
    private Button decreaseElapsedTimeButton;

    private static final Long CHANGE_STEP_ELAPSED_TIME = (long) 1000*60*5;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fill_task_data);

        viewModel = new ViewModelProvider(this).get(TaskStorageViewModel.class);
        intentData = getIntent();

        getFieldsFromForm();
        setListeners();

        if( !intentData.hasExtra("task_id") ) {
            deleteTaskButton.setVisibility(View.INVISIBLE);
            setFillTask(new Task());
        }
        else {
            Consumer<Task> taskConsumer = task -> setFillTask(task);
            viewModel.getTaskByIDAsync(intentData.getLongExtra("task_id", 0), taskConsumer);

        }
    }

    protected void getFieldsFromForm() {
        title = (EditText) findViewById(R.id.edit_text_task_title);
        descr = (EditText) findViewById(R.id.edit_text_task_desctiption);
        priority = (TextView)findViewById(R.id.text_view_priority);
        deleteTaskButton = (ImageButton)findViewById(R.id.delete_task);
        seekBarPriority = (SeekBar) findViewById(R.id.seek_bar_task_priority);

        // fields and buttons for "elapsed time"
        formatElapsedTime = (EditText) findViewById(R.id.elapsed_format_time);
        increaseElapsedTimeButton = (Button) findViewById(R.id.button_increase_elapsed_time);
        decreaseElapsedTimeButton = (Button) findViewById(R.id.button_decrease_elapsed_time);
    }

    protected void setListeners() {
        seekBarPriority.setOnSeekBarChangeListener(new PrioritySetSeekBar());
    }

    public void setFillTask(Task task) {
        fillingTask = task;
        fillFormTask();
    }

    protected void fillFormTask() {
        title.setText(fillingTask.getTitle());
        descr.setText(fillingTask.getDescription());
        priority.setText(fillingTask.getPriority().toString());
        seekBarPriority.setProgress(fillingTask.getPriority());

        Long curTime = new GregorianCalendar().getTimeInMillis();

        formatElapsedTime.setText(TimeFormatter.intervalFromTime(fillingTask.getElapsedTime()));

        if(fillingTask.getStatusId().equals(Task.IN_WORK)) {
            formatElapsedTime.setText(TimeFormatter.intervalFromTime(fillingTask.getElapsedTime() + curTime - fillingTask.getLastUpdateStatus()));
            fillingTask.setElapsedTime(fillingTask.getElapsedTime() + curTime - fillingTask.getLastUpdateStatus());
        }

        fillingTask.setLastUpdateStatus(curTime);
    }

    public void onClickSaveTask(View view) {
        fillingTask.setTitle(title.getText().toString());
        fillingTask.setDescription(descr.getText().toString());
        fillingTask.setPriority(Integer.parseInt(priority.getText().toString()));

        viewModel.SaveTask(fillingTask);

        finish();
    }

    public void onClickBackFromSave(View view) {
        finish();
    }

    public void onClickDeleteTask(View view) {
        viewModel.DeleteTask(fillingTask);
        finish();
    }


    private class PrioritySetSeekBar implements SeekBar.OnSeekBarChangeListener {

        @Override
        public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
            priority.setText(Integer.toString(progress));
        }

        @Override
        public void onStartTrackingTouch(SeekBar seekBar) {

        }

        @Override
        public void onStopTrackingTouch(SeekBar seekBar) {
        }
    }

    public void onClickDecreaseElapsedTime(View view) {
        Long elapsed_time = fillingTask.getElapsedTime();
        if(elapsed_time < CHANGE_STEP_ELAPSED_TIME)
            elapsed_time = (long)0;
        else
            elapsed_time -= CHANGE_STEP_ELAPSED_TIME;


        fillingTask.setElapsedTime(elapsed_time);
        formatElapsedTime.setText(TimeFormatter.intervalFromTime(elapsed_time));
    }

    public void onClickIncreaseElapsedTime(View view) {
        Long elapsed_time = fillingTask.getElapsedTime() + CHANGE_STEP_ELAPSED_TIME;
        fillingTask.setElapsedTime(elapsed_time);
        formatElapsedTime.setText(TimeFormatter.intervalFromTime(elapsed_time));
    }


}

File: app/src/main/java/com/alexpaxom/workrhythm/app/activites/FillTaskDataFormActivity.kt
package com.alexpaxom.workrhythm.app.activites

import androidx.appcompat.app.AppCompatActivity
import com.alexpaxom.workrhythm.app.viewmodel.TaskStorageViewModel
import com.google.android.material.textfield.TextInputEditText
import android.widget.TextView
import android.content.Intent
import android.widget.ImageButton
import android.widget.SeekBar
import android.os.Bundle
import android.view.View
import android.widget.Button
import com.alexpaxom.workrhythm.R
import androidx.lifecycle.ViewModelProvider
import com.alexpaxom.workrhythm.helpers.TimeFormatter
import android.widget.SeekBar.OnSeekBarChangeListener
import androidx.core.util.Consumer
import com.alexpaxom.workrhythm.data.model.Task
import java.util.*

class FillTaskDataFormActivity : AppCompatActivity() {
    private var viewModel: TaskStorageViewModel? = null
    private var title: TextInputEditText? = null
    private var description: TextInputEditText? = null
    private var priority: TextView? = null
    private var fillingTask: Task? = null
    private var deleteTaskButton: ImageButton? = null
    private var seekBarPriority: SeekBar? = null
    private var formatElapsedTime: TextInputEditText? = null
    private var increaseElapsedTimeButton: Button? = null
    private var decreaseElapsedTimeButton: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_fill_task_data)
        viewModel = ViewModelProvider(this).get(TaskStorageViewModel::class.java)
        fieldsFromForm
        setListeners()
        if (!intent.hasExtra("task_id")) {
            deleteTaskButton!!.visibility = View.INVISIBLE
            setFillTask(Task())
        } else {
            val taskConsumer = Consumer { task: Task? -> setFillTask(task) }
            viewModel!!.getTaskByIDAsync(intent.getLongExtra("task_id", 0), taskConsumer)
        }
    }

    // fields and buttons for "elapsed time"
    protected val fieldsFromForm: Unit
        protected get() {
            title = findViewById<View>(R.id.edit_text_task_title) as TextInputEditText
            description = findViewById<View>(R.id.edit_text_task_description) as TextInputEditText
            priority = findViewById<View>(R.id.text_view_priority) as TextView
            deleteTaskButton = findViewById<View>(R.id.delete_task) as ImageButton
            seekBarPriority = findViewById<View>(R.id.seek_bar_task_priority) as SeekBar

            // fields and buttons for "elapsed time"
            formatElapsedTime =
                findViewById<View>(R.id.edit_text_elapsed_hours) as TextInputEditText
            increaseElapsedTimeButton =
                findViewById<View>(R.id.button_increase_elapsed_time) as Button
            decreaseElapsedTimeButton =
                findViewById<View>(R.id.button_decrease_elapsed_time) as Button
        }

    protected fun setListeners() {
        seekBarPriority!!.setOnSeekBarChangeListener(PrioritySetSeekBar())
    }

    fun setFillTask(task: Task?) {
        fillingTask = task
        runOnUiThread { fillFormTask() }
    }

    protected fun fillFormTask() {
        title!!.setText(fillingTask!!.title)
        description!!.setText(fillingTask!!.description)
        priority!!.text = fillingTask!!.priority.toString()
        seekBarPriority!!.progress = fillingTask!!.priority
        val curTime = GregorianCalendar().timeInMillis
        formatElapsedTime!!.setText(TimeFormatter.intervalFromTime(fillingTask!!.elapsedTime))
        if (fillingTask!!.statusId == Task.IN_WORK) {
            formatElapsedTime!!.setText(TimeFormatter.intervalFromTime(fillingTask!!.elapsedTime + curTime - fillingTask!!.lastUpdateStatus))
            fillingTask!!.elapsedTime =
                fillingTask!!.elapsedTime + curTime - fillingTask!!.lastUpdateStatus
        }
        fillingTask!!.lastUpdateStatus = curTime
    }

    fun onClickSaveTask(view: View?) {
        fillingTask!!.title = title!!.text.toString()
        fillingTask!!.description = description!!.text.toString()
        fillingTask!!.priority = priority!!.text.toString().toInt()
        viewModel!!.SaveTask(fillingTask!!)
        finish()
    }

    fun onClickBackFromSave(view: View?) {
        finish()
    }

    fun onClickDeleteTask(view: View?) {
        viewModel!!.DeleteTask(fillingTask!!)
        finish()
    }

    private inner class PrioritySetSeekBar : OnSeekBarChangeListener {
        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
            priority!!.text = Integer.toString(progress)
        }

        override fun onStartTrackingTouch(seekBar: SeekBar) {}
        override fun onStopTrackingTouch(seekBar: SeekBar) {}
    }

    fun onClickDecreaseElapsedTime(view: View?) {
        var elapsed_time = fillingTask!!.elapsedTime
        if (elapsed_time < CHANGE_STEP_ELAPSED_TIME) elapsed_time =
            0.toLong() else elapsed_time -= CHANGE_STEP_ELAPSED_TIME
        fillingTask!!.elapsedTime = elapsed_time
        formatElapsedTime!!.setText(TimeFormatter.intervalFromTime(elapsed_time))
    }

    fun onClickIncreaseElapsedTime(view: View?) {
        val elapsed_time = fillingTask!!.elapsedTime + CHANGE_STEP_ELAPSED_TIME
        fillingTask!!.elapsedTime = elapsed_time
        formatElapsedTime!!.setText(TimeFormatter.intervalFromTime(elapsed_time))
    }

    companion object {
        private const val CHANGE_STEP_ELAPSED_TIME = 1000.toLong() * 60 * 5
    }
}